komodo-relational-commands/src/main/java/org/komodo/relational/commands/server/ServerManager.java

```diff
@@ -0,0 +1,219 @@
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301 USA.
+ */
+package org.komodo.relational.commands.server;
+
+import javax.jcr.Node;
+import javax.jcr.Session;
+import org.komodo.core.KomodoLexicon;
+import org.komodo.relational.teiid.Teiid;
+import org.komodo.repository.ObjectImpl;
+import org.komodo.repository.RepositoryImpl;
+import org.komodo.repository.RepositoryImpl.UnitOfWorkImpl;
+import org.komodo.shell.api.WorkspaceStatus;
+import org.komodo.spi.KException;
+import org.komodo.spi.constants.StringConstants;
+import org.komodo.spi.repository.KomodoObject;
+import org.komodo.spi.repository.Repository.UnitOfWork;
+import org.komodo.spi.runtime.TeiidInstance;
+import org.modeshape.jcr.api.JcrTools;
+
+/**
+ *  Manages the default server used for the VDB Builder Session.
+ */
+public class ServerManager implements StringConstants {
+
+    /**
+     * The root path of the Komodo servers area at the environment root.
+     */
+    public static String SERVERS_ROOT = ( RepositoryImpl.ENV_ROOT + StringConstants.FORWARD_SLASH + KomodoLexicon.Environment.SERVERS );
+
+    /**
+     * Global property key - indicates if server connection should be attempted on startup.
+     */
+    public static String SERVER_CONNECT_ON_STARTUP = "SERVER_CONNECT_ON_STARTUP"; //$NON-NLS-1$
+
+    /**
+     * Global property key - indicates frequency of server connection status check.
+     */
+    public static String SERVER_STATUS_CHECK_SEC = "SERVER_STATUS_CHECK_SEC"; //$NON-NLS-1$
+
+    /**
+     * The name of the default server
+     */
+    public static final String DEFAULT_SERVER_NAME = "DefaultServer"; //$NON-NLS-1$
+
+    /**
+     * The default server connect on startup value
+     */
+    public static final boolean DEFAULT_SERVER_CONNECT_ON_STARTUP = true;
+
+    /**
+     * The default server connect on startup value
+     */
+    public static final int DEFAULT_SERVER_STATUS_CHECK_SEC = 5;
+    
+
+    private static ServerManager instance;
+    private WorkspaceStatus wsStatus;
+    private Teiid server;
+    private boolean isConnected = false;
+    private long lastConnectTestMillis = 0;
+    
+    /**
+     * @param wsStatus the WorkspaceStatus
+     * @return singleton instance
+     */
+    public static ServerManager getInstance( final WorkspaceStatus wsStatus ) {
+        if (instance == null) {
+            instance = new ServerManager();
+        }
+        instance.setWorkspaceStatus(wsStatus);
```

Comment:
Should this line be moved up into the above block where the `ServerManager` instance is being created? And maybe just code it like `instance.wsStatus = wsStatus` and get rid of the `setWorkspaceStatus` method. I am assuming we will never have more than one `WorkspaceStatus` during a shell session so it is only the first time when you need to set the status. Wdyt?